app/controllers/api/user_blocks_controller.rb

```diff
@@ -14,5 +16,30 @@ def show
     rescue ActiveRecord::RecordNotFound
       raise OSM::APINotFoundError
     end
+
+    def create
+      raise OSM::APIBadUserInput, "No user was given" unless params[:user]
+
+      user = User.visible.find_by(:id => params[:user])
+      raise OSM::APINotFoundError unless user
+      raise OSM::APIBadUserInput, "No reason was given" unless params[:reason]
+      raise OSM::APIBadUserInput, "No period was given" unless params[:period]
+
+      period = Integer(params[:period], :exception => false)
+      raise OSM::APIBadUserInput, "Period one is in a wrong format" unless period
```

Comment:
I'm not sure what `one` is referring to in this message? It should also be `the wrong format` but overall I'd probably replace the whole thing with something like `Period should be a number of hours`.

Reply:
copy-paste accident, fixed

Reply:
I was copying the structure of error messages from other controllers where you can find "Date #{params[:from]} is in a wrong format".

Changed to "Period should be a number of hours".